apps/studio/src/server/modules/page/page.router.ts

```diff
@@ -7,18 +10,44 @@ import {
   updatePageSchema,
 } from "~/schemas/page"
 import { protectedProcedure, publicProcedure, router } from "~/server/trpc"
+import { safeJsonParse } from "~/utils/safeJsonParse"
 import {
   getFooter,
   getFullPageById,
   getNavBar,
+  getPageById,
   updateBlobById,
   updatePageById,
 } from "../resource/resource.service"
 import { getSiteConfig } from "../site/site.service"
 
+const typeCompiler = TypeCompiler.Compile(IsomerPageSchema)
+
 // TODO: Need to do validation like checking for existence of the page
 // and whether the user has write-access to said page
 const pageProcedure = protectedProcedure
+const validatedPageProcedure = pageProcedure.use(async ({ next, rawInput }) => {
```

Comment:
nit: you can chain directly without declaring multiple variables

```suggestion
const pageProcedure = protectedProcedure.use(async ({ next, rawInput }) => {
```

Reply:
Hmm, believe that pageProcedure is the generic procedure used without schema validation, while validatedpageprocedure is pageProcedure with a schema validation middleware. Both seems to be used so I am thinking that this is okay?

Reply:
gotcha. in that case maybe don't need to reassign `protectedProcedure` with `pageProcedure` and use `protectedProcedure` straight in non-validated procedures?

Reply:
Okay, might be that I am misunderstanding this, but let me pen down what I think each procedure does?
1. protectedProcedure(declared elsewhere): this has user authentication middleware, used for pages, folders, and everything that isn't public.
2. pageProcedure: currently just protectedProcedure, but eventually when doing up permissioning, will have to add a middleware(so `const pageProcedure = protectedProcedure.use(...)` to enforce permissions, make sure user has perms to CRUD on a page resource(could be folder too, so this procedure might have to shift)
3. validatedPageProcedure: might be bad naming on my part but this is intended to be a pageProcedure with additional middleware for schema val. So what this means is it will have both perms middleware and schema val middleware!

Reply:
Yes, but since currently 1 == 2, it is somewhat redundant. Can always add it in the future. Don't code for the future; code for the present haha.

Reply:
Removed in 49d00f8